runtime/src/bank.rs

```diff
@@ -498,6 +547,95 @@ impl Bank {
         new
     }
 
+    /// Create a bank from explicit arguments and deserialized fields from snapshot
+    pub(crate) fn new_from_fields(
+        bank_rc: BankRc,
+        genesis_config: &GenesisConfig,
+        fields: BankFields,
+    ) -> Self {
+        let mut bank = Self {
+            rc: bank_rc,
+            src: Default::default(),
```

Comment:
@svenski123 Well, this causes our custom sanity CI check failure: https://buildkite.com/solana-labs/solana/builds/25939#1ea7fd6a-912d-4911-8a57-046abaf2a617/78-79. Also, there are several similar ones.

Could you explicitly spell-out the types?

Reply:
I've reviewed #2630 / #2711 and the motivation behind avoiding the use of Default::default() in function calls to improve code readability.

In the struct initialization case here, the member variable names are present and the code readability concerns do not apply in this instance, I have therefore relied on Rust's type inference but in a way that does not violate the CI check.